radiation/python/radiation_driver.py

```diff
@@ -447,73 +597,73 @@ def GFS_radiation_driver(
             k1 = k + kd
             k2 = k + lsk
             for i in range(IM):
-                plvl[i, k1 + kb] = Statein["prsi"][i, k2 + kb] * 0.01  # pa to mb (hpa)
-                plyr[i, k1] = Statein["prsl"][i, k2] * 0.01  # pa to mb (hpa)
-                tlyr[i, k1] = Statein["tgrs"][i, k2]
-                prslk1[i, k1] = Statein["prslk"][i, k2]
+                plvl[i, 0, k1 + kb] = Statein["prsi"][i, 0, k2 + kb] * 0.01  # pa to mb (hpa)
+                plyr[i, 0, k1+1] = Statein["prsl"][i, 0, k2+1] * 0.01  # pa to mb (hpa)
+                tlyr[i, 0, k1+1] = Statein["tgrs"][i, 0, k2+1]
+                prslk1[i, 0, k1+1] = Statein["prslk"][i, 0, k2+1]
```

Comment:
it looks like kd depends on whether the vertical axis is flipped or not (we don't want to support both) and self.lextop,  a control flag for an extra top layer.  lextop = LTP > 0. LTP = 0  # no extra top layer, though and I don't think we need to support both modes at the moment (or maybe ever). lsk might be a little trickier to deal with. Trying to make this a stencil does indeed seem like a task for a future PR> 
l